lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1455,26 +1431,26 @@ def _writePng(self, data):
                 buffer.seek(length, 1)
             buffer.seek(4, 1)   # skip CRC
 
-    def _writeImg(self, data, height, width, grayscale, id, smask=None):
+    def _writeImg(self, data, id, smask=None):
```

Comment:
do we ever actually pass in a grayscale image?

Reply:
We do pass single-channel images (for alpha).  As for actual grayscale I don't know (see https://github.com/matplotlib/matplotlib/pull/12871 for the similar thing in postscript), but this PR doesn't change that behavior anyways.